internal/consensus/state_test.go

```diff
@@ -895,128 +1225,165 @@ func TestStateLockPOLUnlockOnUnknownBlock(t *testing.T) {
-		t.Fatal(err)
-	}
-
-	// more prevote creating a majority on the new block and this is then committed
-	signAddVotes(ctx, t, config, cs1, tmproto.PrecommitType, nil, types.PartSetHeader{}, vs2, vs3, vs4)
+	validatePrecommit(ctx, t, cs1, round, -1, vss[0], nil, nil)
 
-	// before we timeout to the new round set the new proposal
-	cs3 := newState(ctx, t, logger, cs1.state, vs3, kvstore.NewApplication())
-	require.NoError(t, err)
-	prop, propBlock = decideProposal(ctx, t, cs3, vs3, vs3.Height, vs3.Round+1)
-	if prop == nil || propBlock == nil {
-		t.Fatal("Failed to create proposal block with vs2")
-	}
-	thirdPropBlockParts, err := propBlock.MakePartSet(partSize)
-	require.NoError(t, err)
-	thirdPropBlockHash := propBlock.Hash()
-	require.NotEqual(t, secondBlockHash, thirdPropBlockHash)
+	signAddVotes(ctx, t, cs1, tmproto.PrecommitType, config.ChainID(), types.BlockID{}, vs2, vs3, vs4)
 
 	incrementRound(vs2, vs3, vs4)
 
 	// timeout to new round
 	ensureNewTimeout(t, timeoutWaitCh, height, round, cs1.config.Precommit(round).Nanoseconds())
 
-	round++ // moving to the next round
-	ensureNewRound(t, newRoundCh, height, round)
-	t.Log("### ONTO ROUND 2")
-
 	/*
-		Round2 (vs3, C) // C C C C // C nil nil nil)
+		Round 1:
+		No proposal new proposal is created.
+		Send a prevote for B, the block from round 0, from each of the validators to cs1.
+		Send a precommit for nil from all of the validators to cs1.
+		cs1 saw a POL for the block it saw in round 0. We ensure that it does not
+		lock on this block, since it did not see a proposal for it in this round.
 	*/
-
-	if err := cs1.SetProposalAndBlock(ctx, prop, propBlock, thirdPropBlockParts, "some peer"); err != nil {
-		t.Fatal(err)
-	}
+	t.Log("### Starting Round 1")
+	round++
+	ensureNewRound(t, newRoundCh, height, round)
 
 	ensurePrevote(t, voteCh, height, round)
-	// we are no longer locked to the first block so we should be able to prevote
-	validatePrevote(ctx, t, cs1, round, vss[0], thirdPropBlockHash)
+	validatePrevote(ctx, t, cs1, round, vss[0], nil) // All validators prevote for the old block.
 
-	signAddVotes(ctx, t, config, cs1, tmproto.PrevoteType, thirdPropBlockHash, thirdPropBlockParts.Header(), vs2, vs3, vs4)
+	// All validators prevote for the old block.
+	signAddVotes(ctx, t, cs1, tmproto.PrevoteType, config.ChainID(), firstBlockID, vs2, vs3, vs4)
 
+	// Make sure that cs1 did not lock on the block since it did not receive a proposal for it.
 	ensurePrecommit(t, voteCh, height, round)
-	// we have a majority, now vs1 can change lock to the third block
-	validatePrecommit(ctx, t, cs1, round, round, vss[0], thirdPropBlockHash, thirdPropBlockHash)
+	validatePrecommit(ctx, t, cs1, round, -1, vss[0], nil, nil)
 }
 
 // 4 vals
 // a polka at round 1 but we miss it
 // then a polka at round 2 that we lock on
 // then we see the polka from round 1 but shouldn't unlock
-func TestStateLockPOLSafety1(t *testing.T) {
+func TestStateLock_POLSafety1(t *testing.T) {
```

Comment:
I might s/POL/POLKA/ .. the two characters are cheap and the readability when something fails is good.

Reply:
POL is used all throughout the `state.go` code to identify the round, I'd prefer to keep it as the initialism that it is to be consistent here.